Makefile.one

```diff
@@ -0,0 +1,443 @@
+	$(CMD_INSTALL) -m 0640 $(OUTPUT_DIR)/libbpf/bpf/*.h $@
+	$(CMD_INSTALL) -m 0640 $(TRACEE_EBPF_OBJ_SRC) $@
+
+#
+# libbpf
+#
+
+LIBBPF_CFLAGS =
+LIBBPF_LDLAGS =
+LIBBPF_SRC = ./tracee-ebpf/3rdparty/libbpf/src
+
+$(OUTPUT_DIR)/libbpf/libbpf.a: $(LIBBPF_SRC) $(wildcard $(LIBBPF_SRC)/*.[ch]) | $(OUTPUT_DIR)
+	CC="$(CMD_CLANG)" \
+		CFLAGS="$(LIBBPF_CFLAGS)" \
+		LD_FLAGS="$(LIBBPF_LDFLAGS)" \
+		$(NORM_MAKE) -C $(LIBBPF_SRC) \
+		BUILD_STATIC_ONLY=1 \
+		DESTDIR=$(abspath ./$(OUTPUT_DIR)/libbpf/) \
+		OBJDIR=$(abspath ./$(OUTPUT_DIR)/libbpf/obj) \
+		INCLUDEDIR= LIBDIR= UAPIDIR= prefix= libdir= \
+		install install_uapi_headers
+
+$(LIBBPF_SRC):
+ifeq ($(wildcard $@), )
+	$(CMD_GIT) submodule update --init --recursive
+endif
+
+#
+# non co-re ebpf
+#
+
+KERN_RELEASE ?= $(UNAME_R)
+KERN_BUILD_PATH ?= $(if $(KERN_HEADERS),$(KERN_HEADERS),/lib/modules/$(KERN_RELEASE)/build)
+KERN_SRC_PATH ?= $(if $(KERN_HEADERS),$(KERN_HEADERS),$(if $(wildcard /lib/modules/$(KERN_RELEASE)/source),/lib/modules/$(KERN_RELEASE)/source,$(KERN_BUILD_PATH)))
+
+NONCORE_TAG = $(subst .,_,$(KERN_RELEASE)).$(subst .,_,$(VERSION))
+
+.PHONY: non-core
+non-core: $(OUTPUT_DIR)/tracee.bpf.$(NONCORE_TAG).o
+
+$(OUTPUT_DIR)/tracee.bpf.$(NONCORE_TAG).o: $(OUTPUT_DIR)/tracee.bpf $(TRACEE_EBPF_OBJ_SRC)
+	$(CMD_CLANG) -S -nostdinc \
+		-D__TARGET_ARCH_$(LINUX_ARCH) \
+		-D__BPF_TRACING__ \
+		-D__KERNEL__ \
+		-include $(KERN_SRC_PATH)/include/linux/kconfig.h \
+		-I $(KERN_SRC_PATH)/arch/$(LINUX_ARCH)/include \
+		-I $(KERN_SRC_PATH)/arch/$(LINUX_ARCH)/include/uapi \
+		-I $(KERN_BUILD_PATH)/arch/$(LINUX_ARCH)/include/generated \
+		-I $(KERN_BUILD_PATH)/arch/$(LINUX_ARCH)/include/generated/uapi \
+		-I $(KERN_SRC_PATH)/include \
+		-I $(KERN_BUILD_PATH)/include \
+		-I $(KERN_SRC_PATH)/include/uapi \
+		-I $(KERN_BUILD_PATH)/include/generated \
+		-I $(KERN_BUILD_PATH)/include/generated/uapi \
+		-I$(OUTPUT_DIR)/tracee.bpf \
+		-Wunused \
+		-Wall \
+		-Wno-frame-address \
+		-Wno-unused-value \
+		-Wno-unknown-warning-option \
+		-Wno-pragma-once-outside-header \
+		-Wno-pointer-sign \
+		-Wno-gnu-variable-sized-type-not-at-end \
+		-Wno-deprecated-declarations \
+		-Wno-compare-distinct-pointer-types \
+		-Wno-address-of-packed-member \
+		-fno-stack-protector \
+		-fno-jump-tables \
+		-fno-unwind-tables \
+		-fno-asynchronous-unwind-tables \
+		-xc -O2 -g -emit-llvm \
+		-c $(TRACEE_EBPF_OBJ_SRC) \
+		-o $(@:.o=.ll)
+	$(CMD_LLC) \
+		-march=bpf -mcpu=$(BPF_VCPU) \
+		-filetype=obj \
+		-o $@ \
+		$(@:.o=.ll)
+	$(CMD_RM) $(@:.o=.ll)
+
+.PHONY: clean-non-core
+clean-non-core:
+	$(CMD_RM) -rf $(OUTPUT_DIR)/tracee.bpf.$(NONCORE_TAG).o
+
+#
+# co-re ebpf
+#
+
+TRACEE_EBPF_OBJ_SRC = ./tracee-ebpf/tracee/tracee.bpf.c
+
+.PHONY: core
+core: $(OUTPUT_DIR)/tracee.bpf.core.o
+
+$(OUTPUT_DIR)/tracee.bpf.core.o: $(OUTPUT_DIR)/tracee.bpf $(TRACEE_EBPF_OBJ_SRC)
+	$(CMD_CLANG) \
+		-D__TARGET_ARCH_$(LINUX_ARCH) \
+		-D__BPF_TRACING__ \
+		-DCORE \
+		-I./tracee-ebpf/tracee/co-re/ \
```

Comment:
should we include this in the `tracee.bpf` dir?

Reply:
We're not rebuilding CO-RE src from the bundle, during runtime, so I guess not. WDYT ?